Move focus to main window of process if it's already running

jonstodle opened the pull request:
This is just the initial work, but I want to get your feedback before I might head far off in the wrong direction.

This code will move focus to the first process which has the same file name as the `FileResult.Description`. If the window is minimized, it will restore it.

I haven't checked if this will work with UWP apps yet. I started by checking if I could get the easy problem to work first.

This will close #38

patriksvensson committed:
Add file indexing settings.

* Set folders to index and exclude from indexing.
* Allow indexing of specific extensions only.
* Ignore hidden files and system files.

Closes #43

jonstodle committed:
Move focus to main window of process if it's already running

patriksvensson commented on src/Jarvis.Addin.Files/FileProvider.cs:
```diff
@@ -55,7 +59,33 @@ protected override Task ExecuteAsync(FileResult result)
                 if (result.Path.Scheme == "shell")
                 {
-                    Process.Start(path);
+                    var descriptionPath = result.Description.Replace("/", "\\");
```
Using the `FileResult.Description` only works in some cases. Where a `lnk`-file has been indexed, this will be the shortcut's description.

jonstodle replied:
Yes, that's what I feared.

patriksvensson commented on src/Jarvis.Core/Interop/Kernel32.cs:
```diff
@@ -0,0 +1,15 @@
+
+namespace Jarvis.Core.Interop
+{
+    public static class Kernel32
```
Would prefer to put this in `public static partial class Win32` and name the file `Win32.Process` or similar.

patriksvensson reviewed (changes requested):
@jonstodle Left some comments.

punker76 committed:
Fix ClearType for TextBox and TextBlock
  This issue is caused by `AllowsTransparency="True"`
  More info: https://blog.magnusmontin.net/2016/07/07/enabling-cleartype-on-a-textbox-in-a-transparent-wpf-window/

punker76 committed:
Add hidden visibility window settings to DisplayRootViewFor
  to prevent flickering the window on startup

patriksvensson committed:
Added option to install update from within Jarvis.

Closes #53

jonstodle committed:
Move kernel32.dll imports to Win32

jonstodle committed:
Add reference to original indexed entry

jonstodle committed:
Use TargetPath to find correct file for lnks

jonstodle commented:
Well.. That's not exactly what I intended to do. Sorry about this. I'm not too experienced at PRs on Github 😞 

I have made a few changes. I'm most curious what you think about the change in `FileResult`: am I "breaking a boundry" by referring back to the original `IdexedEntry`?

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/spectresystems/jarvis?pullRequest=49) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/spectresystems/jarvis?pullRequest=49) before we can accept your contribution.<br/>**1** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: patriksvensson<br/>:x: punker76<br/>:x: jonstodle<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/spectresystems/jarvis?pullRequest=49) it.</sub>
